src/Explorer/Tabs/ScriptTabBase.ts

```diff
@@ -267,22 +267,8 @@ export default abstract class ScriptTabBase extends TabsBase implements ViewMode
       return false;
     }
 
-    const invalidStartCharacters = /^[/?#\\]/;
-    if (invalidStartCharacters.test(id)) {
-      return false;
-    }
-
-    const invalidMiddleCharacters = /^.+[/?#\\]/;
-    if (invalidMiddleCharacters.test(id)) {
-      return false;
-    }
-
-    const invalidEndCharacters = /.*[/?#\\ ]$/;
-    if (invalidEndCharacters.test(id)) {
-      return false;
-    }
-
-    return true;
+    const validIdRegex = /[^\/?#\\]*[^\/?# \\]/;
```

Comment:
I'm wondering if we should put this method in a util and write some unit tests around it catching invalid characters.